x-pack/platform/plugins/shared/alerting/common/routes/schedule/validation/validation_schedule/v1.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { DURATION_REGEX, INTERVAL_FREQUENCY_REGEXP } from '../../constants';
+
+export const validateSchedule = (schedule: {
+  duration: string;
+  recurring?: { every?: string };
+}) => {
+  const { duration, recurring } = schedule;
+  if (recurring?.every && duration !== '-1') {
+    const [, interval, frequency] = recurring?.every?.match(INTERVAL_FREQUENCY_REGEXP) ?? [];
+    const [, durationNumber, durationUnit] = duration.match(DURATION_REGEX) ?? [];
+
+    const intervalInDays = moment
+      .duration(interval, frequency as moment.unitOfTime.DurationConstructor)
+      .asDays();
+
+    const durationInDays = moment
+      .duration(durationNumber, durationUnit as moment.unitOfTime.DurationConstructor)
+      .asDays();
+
+    if (intervalInDays && interval && durationInDays >= intervalInDays) {
+      return `Recurrence every ${recurring?.every} must be longer than the snooze duration ${duration}`;
+    }
+  }
+
+  if (duration === '-1' && recurring) {
```

Comment:
I missed this discussion, but why did we go with `duration` being `-1` instead of `undefined`?

Reply:
@cnasikas What do you think of having `duration: undefined` instead of `duration: -1`? Isn't that clearer?

Reply:
We removed indefinite snooze option (duration = '-1') from this api, as it is handled bit differently as of now and needs some refactoring in server and UI side. It will be handled in a different issue.